File: rest-service/db/src/main/java/mb/lib/db/model/FullJobRow.java
package mb.lib.db.model;

import java.io.File;

public interface FullJobRow extends ShortJobRow, AutoCloseable
{
  /**
   * @return the blast tool configuration submitted for this job.
   */
  String config();

  /**
   * @return the raw query submitted for this job.
   */
  File query();

  @Override
  default void close() throws Exception {
    if (query() != null)
      query().delete();
  }
}

File: rest-service/src/main/kotlin/mb/lib/db/model/FullJobRow.kt
package mb.lib.db.model

import java.io.File

interface FullJobRow: ShortJobRow, AutoCloseable
{
  /**
   * The blast tool configuration submitted for this job.
   */
  val config: String

  /**
   * The raw query submitted for this job.
   */
  val query: File

  override fun close() {
    query.delete()
  }
}
